File: TS/calculator.ts
import * as readline from "node:readline/promises";
import { stdin as input, stdout as output } from "node:process";
import * as fs from "node:fs/promises";

const rl = readline.createInterface({ input, output });
let e = true;
async function calculator() {
  const choice = await rl.question(
    "What would you like to do?\n1.Add\n2.Subtract\n3.Divide\n4.Multiply\n5.Exit\n"
  );

  if (choice === "1") {
    const num1 = await rl.question("What is the first number? ");
    const num2 = await rl.question("What is the second number? ");
    console.log(Number(num1) + Number(num2));
  } else if (choice === "2") {
    const num1 = await rl.question("What is the first number? ");
    const num2 = await rl.question("What is the second number? ");
    console.log(Number(num1) - Number(num2));
  } else if (choice === "3") {
    const num1 = await rl.question("What is the first number? ");
    const num2 = await rl.question("What is the second number? ");
    console.log(Number(num1) / Number(num2));
  } else if (choice === "4") {
    const num1 = await rl.question("What is the first number? ");
    const num2 = await rl.question("What is the second number? ");
    console.log(Number(num1) * Number(num2));
  } else if (choice === "5") {
    e = false;
  } else {
    console.log("Invalid choice");
  }
}
(async () => {
  while (e == true) {
    await calculator();
  }
  rl.close();
})();

File: C++/calculator.cpp
#include <iostream>
#include <windows.h>

using namespace std;

void calculator()
{
    float a;
    float b;
    int menuOption;
    cout << "Welcome\n1.Addition\n2.Subtraction\n3.Division\n4.Multiplication";
    cout << "\n";
    cin >> menuOption;
    if (menuOption == 1)
    {
        cout << "Enter the first number to add: ";
        cin >> a;
        cout << "Enter the second number to add: ";
        cin >> b;
        Sleep(500);
        float ca = a + b;
        cout << ca << "\n";
        Sleep(500);
        calculator();
    }
    else if (menuOption == 2)
    {
        cout << "Enter the first number to subtract: ";
        cin >> a;
        cout << "Enter the second number to subtract: ";
        cin >> b;
        Sleep(500);
        float ca = a - b;
        cout << ca << "\n";
        Sleep(500);
        calculator();
    }
    else if (menuOption == 3)
    {
        cout << "Enter the first number to divide: ";
        cin >> a;
        cout << "Enter the second number to divide: ";
        cin >> b;
        Sleep(500);
        float ca = a / b;
        cout << ca << "\n";
        Sleep(500);
        calculator();
    }
    else if (menuOption == 4)
    {
        cout << "Enter the first number to multiply: ";
        cin >> a;
        cout << "Enter the second number to multiply: ";
        cin >> b;
        Sleep(500);
        float ca = a * b;
        cout << ca << "\n";
        Sleep(500);
        calculator();
    } else {
        cout << "Invalid option" << "\n";
        Sleep(500);
        calculator();
    }
}

int main()
{
    calculator();
}